pkg/templates/templates.go

```diff
@@ -182,6 +182,9 @@ type TemplateResolver struct {
 	// If caching is disabled, this will act as a temporary cache for objects during the execution of the
 	// ResolveTemplate call.
 	tempCallCache client.ObjectCache
+	// Use in template resolver CLI to create outputs including resources used in fromSecret, fromConfigMap,
+	// loopUp etc functions.
+	UsedResources []unstructured.Unstructured
```

Comment:
Small request: can this be unexported (like `usedResources`), and have a `GetUsedResources()` function to access a *copy* of it? I'm just worried because every other field is unexported, it would be nice to keep things consistent.